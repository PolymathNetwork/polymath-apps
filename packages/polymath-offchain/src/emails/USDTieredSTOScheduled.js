// @flow

import React from 'react';
import moment from 'moment';

import { EmailWrapper, EtherscanURL } from './EmailWrapper';
import { POLYMATH_ISSUER_URL, POLYMATH_OFFCHAIN_URL } from '../constants';

type Props = {|
  ticker: string,
  start: Date,
  fundsReceiver: string,
  txHash: string,
  networkId: string,
|};

export const USDTieredSTOScheduled = ({
  ticker,
  start,
  fundsReceiver,
  txHash,
  networkId,
}: Props) => (
  <EmailWrapper>
    <h4>Congratulations!</h4>
    <h1>
      You Have Successfully Created and Scheduled the Security Token Offering
      for {ticker.toUpperCase()}
    </h1>
    <div className="icon-text tx-hash">
      <div className="icon">
        <img alt="Icon" src={`${POLYMATH_OFFCHAIN_URL}/img/checkmark.png`} />
      </div>
      <h2>You can view the transaction details here:</h2>
      <div className="tx">
        Transaction details on Etherscan:{' '}
        <EtherscanURL type={'tx'} hash={txHash} networkId={networkId} />
      </div>
    </div>
    <div className="text">
      <h3>Additional details of your Security Token Offering are below:</h3>
      <div className="value">
        <strong>Scheduled start</strong>
        <p>{moment(start).format('MM/DD/YYYY')}</p>
      </div>
      <div className="value">
        <strong>ETH Address to receive the funds raised during the STO</strong>
        <p>
          <EtherscanURL
            type={'address'}
            hash={fundsReceiver}
            networkId={networkId}
          />
        </p>
      </div>
      <p align="center">
        <a href={`${POLYMATH_ISSUER_URL}/dashboard/${ticker}/sto`}>
          <strong>
            Click here to access your Security Token Offering Dashboard
          </strong>
        </a>
      </p>
    </div>
  </EmailWrapper>
);
