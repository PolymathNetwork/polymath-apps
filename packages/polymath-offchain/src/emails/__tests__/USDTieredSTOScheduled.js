import React from 'react';
import renderer from 'react-test-renderer';
import { USDTieredSTOScheduled } from '../USDTieredSTOScheduled';

describe('Component: USDTieredSTOScheduled', () => {
  const ticker = 'SOMETICK';
  const start = 561168000;
  const fundsReceiver = '0xffffffffffffffff';
  const txHash = '0xeeeeeeeeeeeeeeee';
  const networkId = '15';

  test('renders with all valid props', () => {
    let component = renderer.create(
      <USDTieredSTOScheduled
        ticker={ticker}
        start={start}
        fundsReceiver={fundsReceiver}
        txHash={txHash}
        networkId={networkId}
      />
    );
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
