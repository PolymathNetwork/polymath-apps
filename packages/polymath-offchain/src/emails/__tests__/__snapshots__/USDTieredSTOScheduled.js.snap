// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: USDTieredSTOScheduled renders with all valid props 1`] = `
<html
  lang="en"
>
  <head>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
  body {
    font: small/1.5 -apple-system, BlinkMacSystemFont, Arial, Helvetica, sans-serif;
    color: #152935;
    font-size: 16px;
  }
  
  a {
    color: #3D70B2 !important;
    text-decoration: none;
  }
  
  a:hover {
    text-decoration: underline;
  }
  
  ul,
  ol {
    margin-top: 5px;
  }
  
  ul li,
  ol li {
    font-size: 18px;
    margin-bottom: 7px;
  }
  
  ol li {
    padding-left: 5px;
  }
  
  h1 {
    font-size: 36px;
    line-height: 43px;
    font-weight: 600;
    color: #252E6A;
    margin-top: 0;
    margin-bottom: 11px;
  }
  
  h2 {
    font-size: 20px;
    line-height: 28px;
    color: #152935;
    font-weight: normal;
  }
  
  h3 {
    font-size: 20px;
    font-weight: 600;
    margin-bottom: 12px;
    color: #152935 !important;
  }
  
  h4 {
    color: #3D70B2;
    font-size: 14px;
    line-height: 18px;
    font-weight: 600;
    margin-top: 0;
    margin-bottom: 0;
  }
  
  .wrapper {
    width: 768px;
    margin: 0 auto;
  }
  
  .top-bar {
    height: 48px;
    border-bottom: 1px solid #DFE3E6;
    padding-left: 72px;
    margin-bottom: 24px;
  }
  
  .top-bar img {
    margin-top: 11px;
    width: 188px;
    height: 29px;
  }
  
  .content {
    padding: 3px 72px 25px;
  }
  
  .main-value {
    width: 624px;
    height: 160px;
    background-color: #EBF0F7;
    border-radius: 4px;
    margin-bottom: 32px;
    margin-top: 28px;
  }
  
  .main-value h2 {
    line-height: 18px;
    text-align: center;
    padding-top: 19px;
    font-weight: 600;
  }
  
  .main-value .value {
    width: 560px;
    height: 64px;
    line-height: 64px;
    text-align: center;
    font-size: 20px;
    background-color: white;
    margin: 0 auto;
    vertical-align: middle;
  }
  
  .icon-text {
    margin-bottom: 30px;
  }
  
  .icon-text .icon {
    width: 48px;
    height: 100%;
    float: left;
  }
  
  .icon-text .icon.question {
    margin-top: 5px;
  }
  
  .icon-text .icon img {
    width: 32px;
    height: 32px;
    margin-top: 4px;
  }
  
  .icon-text h2 {
    margin: 0;
    line-height: 25px;
  }
  
  .icon-text p {
    margin-top: 3px;
  }
  
  .icon-text.tx-hash {
    height: 57px;
    margin-top: 20px;
  }
  
  .text {
    margin-bottom: 26px;
    color: #5A6872;
  }
  
  .text p {
    font-size: 18px;
  }
  
  .value {
    margin-top: 13px;
  }
  
  .value p {
    margin-top: 8px;
  }
  
  .value strong {
    color: #152935 !important;
  }
  
  .footer {
    height: 48px;
    border-top: 1px solid #DFE3E6;
    padding-left: 72px;
    padding-right: 72px;
    font-size: 14px;
    color: #5A6872;
    padding-top: 14px;
  }
  
  .footer .left,
  .footer .right {
    float: left;
    width: 50%;
  }
  
  .footer .right {
    text-align: right;
  }
  
  .footer a {
    color: #5A6872 !important;
    margin-left: 40px;
  }
  
  .tx {
    float: left;
    width: 300px;
    height: 25px;
    border-radius: 4px;
    background-color: #EBF0F7;
    font-size: 12px;
    line-height: 25px;
    padding: 0 16px;
    margin-top: 6px;
    text-align: center;
  }
  
  .tx a {
    text-decoration: underline;
  }
  
  .sincere {
    font-weight: 600;
    color: #252E6A;
    line-height: 31px;
  }
",
        }
      }
    />
  </head>
  <body>
    <div
      className="wrapper"
    >
      <div
        className="top-bar"
      >
        <img
          alt="Icon"
          src="http://localhost:3001/img/logo.png"
        />
      </div>
      <div
        className="content"
      >
        <h4>
          Congratulations!
        </h4>
        <h1>
          You Have Successfully Created and Scheduled the Security Token Offering for 
          SOMETICK
        </h1>
        <div
          className="icon-text tx-hash"
        >
          <div
            className="icon"
          >
            <img
              alt="Icon"
              src="http://localhost:3001/img/checkmark.png"
            />
          </div>
          <h2>
            You can view the transaction details here:
          </h2>
          <div
            className="tx"
          >
            Transaction details on Etherscan:
             
            <a
              href="https://local.etherscan.io/tx/0xeeeeeeeeeeeeeeee"
              rel="noopener noreferrer"
              target="_blank"
            >
              0xeeeeeeee...eeee
            </a>
          </div>
        </div>
        <div
          className="text"
        >
          <h3>
            Additional details of your Security Token Offering are below:
          </h3>
          <div
            className="value"
          >
            <strong>
              Scheduled start
            </strong>
            <p>
              10/14/1987
            </p>
          </div>
          <div
            className="value"
          >
            <strong>
              ETH Address to receive the funds raised during the STO
            </strong>
            <p>
              <a
                href="https://local.etherscan.io/address/0xffffffffffffffff"
                rel="noopener noreferrer"
                target="_blank"
              >
                0xffffffffffffffff
              </a>
            </p>
          </div>
          <p
            align="center"
          >
            <a
              href="http://localhost:3000/dashboard/SOMETICK/sto"
            >
              <strong>
                Click here to access your Security Token Offering Dashboard
              </strong>
            </a>
          </p>
        </div>
        <div
          className="icon-text"
          style={
            Object {
              "height": "52px",
            }
          }
        >
          <div
            className="icon question"
          >
            <img
              alt="Icon"
              src="http://localhost:3001/img/question.png"
            />
          </div>
          <h2>
            If you have any questions, please reach out to
            <br />
            <a
              href="mailto:tokenstudio@polymath.zendesk.com"
            >
              tokenstudio@polymath.zendesk.com
            </a>
          </h2>
        </div>
        <h2
          className="sincere"
        >
          Best,
          <br />
          Polymath Support
        </h2>
      </div>
      <div
        className="footer"
      >
        <div
          className="left"
        >
          Â© 2018 Polymath
        </div>
        <div
          className="right"
        >
          <a
            href="https://polymath.network/termsofservice.html"
          >
            Terms of service
          </a>
          <a
            href="https://polymath.network/privacypolicy.html"
          >
            Privacy policy
          </a>
        </div>
      </div>
    </div>
  </body>
</html>
`;
